Add Shared Databases Across Environments to Licensing (#796)
directus/docs: content/licensing/1.overview.md

Set the license through the `LICENSE_KEY` (or `LICENSE_TOKEN`) environment variable at the service or task level so that every replica inherits the same value automatically. New containers spun up by autoscaling pick up the license on start without any manual setup or per-container retrieval.

### Shared Databases Across Environments

> Some deployment pipelines deploy a release to a preprod environment first, then promote the same release to production — both pointed at the same database, but with different `PUBLIC_URL` values.

Reviewer: This reads like a very specific deployment problem. Should it be at the bottom of the page in a troubleshooting section or similar rather than here?

Author: Added a new troubleshooting section in the next commit 👍

Since a license binds to your project (stored in the database) and its `PUBLIC_URL` (set per environment), using a **different `LICENSE_KEY` in each environment** causes the binding to conflict: whichever environment starts second will try to rebind the shared project to its own key and URL, and the licensing service will reject it.

> Use the **same `LICENSE_KEY` value** across every environment that shares a database, varying only `PUBLIC_URL`. This is the same multi-activation model described above — each environment registers its own binding for its own URL, under one key.

Reviewer: This reads confusingly to me... the first sentence almost infers the opposite (it's possible).

If you already hit a binding conflict:

1. On the environment with the working license, remove `LICENSE_KEY` from the environment and restart. This unlocks the Studio license editor.
2. Go to **Settings > License** and click **Deactivate License**. This releases the binding on the licensing service, rather than just clearing it locally.
3. Stop that environment.
4. Start the other environment. With no license data left in the database, it activates fresh.

::callout{icon="i-lucide-triangle-alert" color="warning"}

> Don't clear `license_key` directly in the database to work around a binding conflict. This leaves the old activation stranded on the licensing service, still counted against your activation limit. Always deactivate through the Studio or API first.

Reviewer: Good callout 👍

Set the license through the `LICENSE_KEY` (or `LICENSE_TOKEN`) environment variable at the service or task level so that every replica inherits the same value automatically. New containers spun up by autoscaling pick up the license on start without any manual setup or per-container retrieval.

> Using a different `LICENSE_KEY` per environment against a shared database causes the environments' bindings to conflict. See [Troubleshooting](#troubleshooting) below.

Reviewer: It's very implementation detail specific. Are you sure we don't want to simply say...

```suggestion
Using separate `LICENSE_KEY`s but different `PUBLIC_URL`s with a shared database is not supported. See [Troubleshooting](#troubleshooting) below.
```

- **Above core tier limits** — the instance enters a 30-day grace period during which it continues to function normally. Admins see a reminder on every login. Add a license key to clear the reminder, or reduce usage to within core limits before the grace period ends.

If the grace period ends with usage still above core limits, the resolution flow is triggered on next admin login.

## Troubleshooting

> ### Binding conflicts across shared-database environments

Reviewer: Does a customer know what a binding is? It sounds like internal implementation/AI language?

> If environments that share a database are configured with **different** `LICENSE_KEY` values, whichever environment starts second tries to rebind the shared project to its own key and `PUBLIC_URL`. The licensing service rejects this, and that environment fails to start with a binding error.

Reviewer: `The licensing service rejects this` this is very internal implementation specific language. The docs should be customer-facing language IMO
